### DISCUSSION TITLE
3D panel feature request: enable multiple Publish widgets

### DISCUSSION THREAD

**Discussion by tonynajjar** [foxglove/community#1057] [General]
Coming from Rviz, it would be nice if the Foxglove 3D panel would allow the creation of multiple "Publish" objects/widgets so we don't have to change the topics all the time. One common example in mobile robots is to have one widget for sending a goal and another to send a pose estimate

**Comment by jtbandes** [comment]
Another person asked about this in Discord just last week. https://discord.com/channels/1240935776649478144/1309269038274314367

There are "multiple" publish tools, but they are grouped under the same icon. You have to long press on the icon to switch between them:

![image](https://github.com/user-attachments/assets/2de30ccf-1a2d-4216-ba07-e0d0ccdd532a)

**Reply by tonynajjar** [reply]
Ohh nice I didn't know about that but that still limits us to 1 topic per type right? E.g. A navigation goal and a localization pose estimate are both PoseStamped so I can't easily switch between these

**Reply by jtbandes** [reply]
Does RViz have this ability? My impression was it also had only 3 separate types? (This Foxglove feature was originally modeled off RViz)

**Reply by tonynajjar** [reply]
Yes pretty sure rviz has that ability. You can't specify a topic on the fly but you can pre-code multiple plugins using the same msg type, publishing on different topics.

**Reply by jtbandes** [reply]
Closing this issue out – since it was initially filed we've made some other minor UX improvements to the 3D publish tool. As far as I can tell RViz behaves pretty much the same, where there is one tool of each type.
